packages/new-design/lib/utils.mjs

```diff
@@ -114,41 +117,123 @@ export const getChoices = async () => {
 }
 
 // Keep track of directories that need to be created
-const dirs = {}
+const dirPromises = {}
 const ensureDir = async (file, suppress = false) => {
   const dir = suppress ? dirname(file.replace(suppress)) : dirname(file)
-  if (!dirs[dir]) {
-    await mkdir(dir, { recursive: true })
-    dirs[dir] = true
+  if (!dirPromises[dir]) {
+    dirPromises[dir] = mkdir(dir, { recursive: true })
   }
+  await dirPromises[dir]
 }
 
 // Helper method to copy template files
-const copyTemplate = async (config, choices) => {
-  // Copy files in parallel rather than using await
-  const promises = []
+const copyFileOrTemplate = async (fromRootOrTemplate, toRoot, relativeDest, templateVars) => {
+  const to = join(toRoot, relativeDest)
 
-  // Copy shared files
-  for (const from of config.files.shared) {
-    // FIXME: Explain the -7
-    const to = join(config.dest, from.slice(config.source.shared.length - 7))
-    if (!dirs[to]) await ensureDir(to)
-    promises.push(copyFile(from, to))
+  await ensureDir(to)
+
+  if (templateVars) {
+    const rendered = mustache.render(fromRootOrTemplate, templateVars)
+    await writeFile(to, rendered)
+  } else {
+    const from = join(fromRootOrTemplate, relativeDest)
+    await copyFile(from, to)
+  }
+}
```

Comment:
IMO this function is a little confusing now. `fromRootOrTemplate` may be either:

* A directory name (i.e. a `fromRoot`), in which case `relativeDest` doubles as a `relativeSource`
* The serialized _contents_ of a template file

Might be better for this argument to act as the same kind of thing in both cases, or two split this function into two function, `copyFile` and e.g. `renderTemplate`